### ISSUE TITLE
Support Different Client Certificates per Environment

### ISSUE THREAD

**Issue by the-powerpointer** [usebruno/bruno#3338]
### I have checked the following:

- [X] I've searched existing issues and found nothing related to my issue.

### Describe the feature you want to add

Currently it is nice that one can switch environments with some variables, e.g. URL, credential, etc.
However, client certificates are specified for the whole collection and map to a certain domain.

I have unfortunately a service which has one URL to retrieve a token from and one URL of the service itself.
The latter is no problem, as this URL changes per environment and I can extend the collection to add client certificates for that URL.
The URL where the token is retrieved from, is however shared between many service instances. When requesting a token, the client certificates are evaluated as well is the `clientId` in the parameters, and both have to match.
This means, with the current implementation I can only point to one environment. Changing the environment means to change the `clientId`, but not the client certificate against that authentication URL...

What is needed?
- Specification of client certificates-to-url mapping in each environment
- Still the current mechanism could stay, but it should be searched for client certificates first in the currently selected environment 

### Mockups or Images of the feature

![image](https://github.com/user-attachments/assets/7e936dbd-5926-4b66-8bc3-18134e9af22d)

**Comment by helloanoop** [comment]
Hey @the-powerpointer! Thanks for thoroughly outlining the feature request and your specific use case.

Have you considered using a separate environment variable, like tokenClientId, for the token request? This way, each environment can have distinct client IDs—`clientId` for accessing the resource URL and `tokenClientId` for fetching the token.

**Comment by mattisking** [comment]
You should be able to accomplish this if I'm understanding it correctly by doing this:

![image](https://github.com/user-attachments/assets/34e02cd1-f139-4af8-951d-59212753788a)

![image](https://github.com/user-attachments/assets/fdaf98c1-253e-49f2-85c9-3a668f8f446d)

Then edit the collection settings:
`{
  "version": "1",
  "name": "Demo",
  "type": "collection",
  "ignore": [
    "node_modules",
    ".git"
  ],
  "clientCertificates": {
    "enabled": true,
    "certs": [
      {
        "domain": "localhost:4356",
        "type": "pfx",
        "pfxFilePath": "{{PFX_PATH}}",
        "passphrase": "{{PFX_PF}}"
      }
    ]
  }
}`

Now you can set the PFX_PATH and PFX_PF differently for each environment you want to create. It will send the proper cert despite the domain always being the same because it will use the value from your environment settings. If you mark them as I did in the image as Secrets, then when you check this into Git, you won't be setting the Path and PassPhrase for other developers and they can do this themselves. I've found this very flexible and it's how I setup all my collections. Note that we also use MTLS.

**Comment by mattisking** [comment]
I’m glad to help! We use MTLS heavily and so this was a major need for me as well.

**Comment by horndt-nlcom** [comment]
I would like to add to this request.
In my case we have one url to make requests to per environment, but cert (and key, and passphrase) are the only thing that is checked on the receiving side.

So for the development environment:
https://development.this.api
And if we use cert1-xyz-development.pem we make a request to the xyz client.
If we use cert1-abc-development.pem we make a request to the abc client.

So it would be nice, if we could add multiple certificates for the same host, but then select which one should be active for an environment.

**Comment by mattisking** [comment]
So why not define your environment based on the cert and not the url? 

development.this.api is your url and then
Environment XYZ sends that first cert and ABC sends that second one.

**Comment by horndt-nlcom** [comment]
@mattisking Yeah that works, would be more clear if it would all be in one place imo.
We have about 16 different certificates for the same url, so a lot of environments. (16x3, dev, staging, prod)
